presto-local-file/src/main/java/io/prestosql/plugin/localfile/LocalFileRecordCursor.java

```diff
@@ -246,6 +248,14 @@ public void close()
         reader.close();
     }
 
+    private static long parseTimestamp(String value)
+    {
+        OffsetDateTime time = OffsetDateTime.parse(value, ISO_FORMATTER);
+        long epochMillis = time.toInstant().toEpochMilli();
```

Comment:
Technically, we should round here in case the parsed instant has more precision that milliseconds.

Reply:
for log parsing, the truncation (not rounding) may be a more expected behavior, as this is what log printer does

Reply:
The connector only parses Airlift HTTP logs which use milliseconds, but I can make this round:
```java
OffsetDateTime time = OffsetDateTime.parse(value, ISO_FORMATTER)
        .plus(500, ChronoUnit.MICROS)
        .truncatedTo(ChronoUnit.MILLIS);